### ISSUE TITLE
Escaped URLs from jenkins are escaped again

### ISSUE THREAD

**Issue by medains** [influxdata/telegraf#8028]
Related to https://github.com/influxdata/telegraf/issues/7165

### Relevant telegraf.conf:

```toml
[[inputs.jenkins]]
url = "http://jenkinshost/"
```

### System info:

```
Telegraf 1.15.2 (git: HEAD cd037b49)
```

Jenkins 2.83 (an old host) and 2.235 (latest stable version in redhat)

### Steps to reproduce:

1. Create a jenkins job with a space in the name
2. telegraf --config telegraf.conf --test

### Expected behavior:

```
> jenkins_job,host=blah,name=job\ with\ space,port=9080,result=SUCCESS,source=IP duration=562i,result_code=0i 1598345400000000000
```

### Actual behavior:

```
2020-08-25T09:38:37Z E! [inputs.jenkins] Error in plugin: [/job/job%2520with%2520space/1/api/json] 404 Not Found
```

### Additional info:

Jenkins in this version range presents the pre-escaped URLs for the jobs - resulting in double-escaping the URL.

**Comment by reimda** [comment]
Just checking that I understand the issue- Telegraf requests job urls from a jenkins host.  Previously jenkins returned urls that weren't urlencoded so 7165 added that to telegraf.  Now jenkins returns job urls that are already urlencoded so we're doing it twice?

**Comment by medains** [comment]
That's definitely true for the jenkins versions I listed - I've searched the jenkins changelog and looked through the code, and I can't find the version in which it changed - though the URL is provided by an annotation so a change to a java library could have resulted in the change.

The jenkins plugin may need to PathUnescape before escaping the path to cope with the earlier version behaviour.
If the version the change happened on could be determined, then the X-Jenkins header in the response could be used to determine whether escaping was required.
Alternatively the behaviour could be toggled in configuration - but that seems a bit nasty.
